server/src/main/java/org/eclipse/hono/server/HonoServer.java

```diff
@@ -86,7 +81,7 @@
      */
     @Autowired(required = false)
     public HonoServer setHonoConfiguration(final HonoConfigProperties props) {
```

Comment:
This is exactly what I do not like about it :-) The warning can be output in many other places as well, e.g. when starting up the server(s).
The ability to set the configuration properties is, however, common to and needed by all servers.